elastic/tests/test_config.py

```diff
@@ -86,3 +89,67 @@ def test_from_instance():
     assert c.url == "https://foo.bar:9200"
     assert c.tags == ["url:https://foo.bar:9200"]
     assert c.service_check_tags == ["host:foo.bar", "port:9200"]
+
+
+@pytest.mark.parametrize(
+    'invalid_custom_queries',
+    [
+        [
+            {
+                'endpoint': '',
+                'metrics': [
+                    {
+                        'datadog_metric_name': 'elasticsearch.custom.metric',
+                        'es_metric_name': '_nodes.total',
+                    },
+                ],
+            }
+        ],
+        [
+            {
+                'endpoint': '/_nodes',
+            }
+        ],
+        [{'endpoint': '/_node', 'metrics': []}],
+        [
+            {
+                'endpoint': '/_nodes',
+                'metrics': [
+                    {
+                        'datadog_metric_name': '',
+                        'es_metric_name': '_nodes.total',
+                    },
+                ],
+            }
+        ],
+        [
+            {
+                'endpoint': '/_nodes',
+                'metrics': [
+                    {
+                        'datadog_metric_name': 'elasticsearch.custom.metric',
+                        'es_metric_name': '',
+                    },
+                ],
+            }
+        ],
+        [
+            {
+                'endpoint': '/_nodes',
+                'metrics': [
+                    {
+                        'datadog_metric_name': 'elasticsearch.custom.metric',
+                    },
+                ],
+            }
+        ],
+    ],
+)
+@pytest.mark.integration
+def test_custom_query_invalid_config(dd_run_check, instance, aggregator, invalid_custom_queries):
+    instance = deepcopy(instance)
+    instance['custom_queries'] = invalid_custom_queries
+    check = ESCheck('elastic', {}, instances=[instance])
+
+    with pytest.raises(Exception):
```

Comment:
Can you test the exact Exception- I think it would be config exception.

Reply:
I wasn't able to get `ConfigurationError` even though the tests would fail with `ConfigurationError`. I don't think the validators are able to return the exact error yet.

Reply:
What error do you see?

Reply:
Here's the full error:
```
tests/test_config.py:157: in test_custom_query_invalid_config
    dd_run_check(check)
../datadog_checks_dev/datadog_checks/dev/plugin/pytest.py:219: in run_check
    raise Exception('\n'.join(exc_lines))
E   Exception: 
E   Traceback (most recent call last):
E     File "/Users/andrew.zhang/go/src/github.com/DataDog/integrations-core/datadog_checks_base/datadog_checks/base/checks/base.py", line 1008, in run
E       initialization()
E     File "/Users/andrew.zhang/go/src/github.com/DataDog/integrations-core/datadog_checks_base/datadog_checks/base/checks/base.py", line 423, in load_configuration_models
E       instance_config = self.load_configuration_model(package_path, 'InstanceConfig', raw_instance_config)
E     File "/Users/andrew.zhang/go/src/github.com/DataDog/integrations-core/datadog_checks_base/datadog_checks/base/checks/base.py", line 463, in load_configuration_model
E       raise_from(ConfigurationError('\n'.join(message_lines)), None)
E     File "<string>", line 3, in raise_from
E   datadog_checks.base.errors.ConfigurationError: Detected 1 error while loading configuration model `InstanceConfig`:
E   __root__
E     Each custom query must have a `datadog_metric_name` and `es_metric_name` value
```

It looks like the actual exception type changes to a generic `Exception` even though it should be a `ConfigurationError`.

Reply:
Ah I see that here https://github.com/DataDog/integrations-core/blob/master/datadog_checks_dev/datadog_checks/dev/plugin/pytest.py#L219

Reply:
that's fine then 👍